src/Repository/ConnectedGithubClient.php

```diff
@@ -0,0 +1,195 @@
+use Github\Exception\RuntimeException;
+use OAT\DependencyResolver\Repository\Entity\Repository;
+use OAT\DependencyResolver\Repository\Exception\BranchNotFoundException;
+use OAT\DependencyResolver\Repository\Exception\EmptyRepositoryException;
+use OAT\DependencyResolver\Repository\Exception\FileNotFoundException;
+
+class ConnectedGithubClient
+{
+    /** @var GithubClientProxy */
+    protected $client;
+
+    /** @var string */
+    protected $organization = '';
+
+    /** @var string */
+    protected $token = '';
+
+    /**
+     * Is the client authenticated?
+     *
+     * @var bool
+     */
+    protected $authenticated = false;
+
+    /**
+     * Stores the numbers of public and private repositories.
+     *
+     * @var array
+     */
+    protected $organisationProperties = [];
+
+    public function __construct(GithubClientProxy $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Stores token for github client.
+     */
+    public function setToken(string $token)
+    {
+        $this->token = $token;
+    }
+
+    public function getOrganizationProperties(string $owner): array
+    {
+        $this->authenticateAndCheck($owner);
+
+        return $this->organisationProperties;
+    }
+
+    /**
+     * Retrieves organization's repositories.
+     *
+     * @param string $owner
+     * @param int    $perPage Number of repositories to load per client call.
+     *
+     * @return array
+     */
+    public function getRepositoryList(string $owner, int $perPage = 100): array
+    {
+        $this->authenticateAndCheck($owner);
+
+        $repositories = [];
+        $page = 0;
+        do {
+            $page++;
+            $foundRepositories = $this->client->getRepositoryList($owner, $page, $perPage);
+            foreach ($foundRepositories as $repository) {
+                $repositoryName = $repository['name'];
+                $repositories[$owner . '/' . $repositoryName] = new Repository(
+                    false, // repository not yet analyzed.
+                    $owner,
+                    $repositoryName,
+                    $repository['private'],
+                    $repository['default_branch']
+                );
+            }
+        } while (count($foundRepositories) === $perPage);
+
+        return $repositories;
+    }
+
+    /**
+     * @throws FileNotFoundException when the file does not exist
+     * @throws ErrorException when another error occurs
+     */
+    public function getContents(string $owner, string $repositoryName, string $branchName, string $fileName): ?string
+    {
+        $this->authenticateAndCheck($owner);
+
+        // Gets a reference to the given branch of the repository if it exists.
+        $branchReference = $this->getBranchReference($owner, $repositoryName, $branchName);
+
+        try {
+            return $this->client->getFileContents($owner, $repositoryName, $branchReference, $fileName);
+        } catch (RuntimeException $exception) {
```

Comment:
According to the doc the client cannot throw a runtime exception. It throws ErrorException

Reply:
You're right. And also use InvalidArgumentException. Fixed.